internal/core/ports/wallet.go

```diff
@@ -85,8 +85,8 @@ type TransactionManager interface {
 }
 
 type NotificationManager interface {
-	TxChannel() chan TxNotification
-	UtxoChannel() chan UtxoNotification
+	TxChannel() (chan TxNotification, error)
+	UtxoChannel() (chan UtxoNotification, error)
```

Comment:
Better adding an `Error() error` method to both `TxNotification` and `UtxoNotification` interfaces